[1.21.10] Crash between Architectury 18.0.6 and Neoforge 21.10.38-beta

Changes in Neoforge 21.10.35-beta cause another incompatibility.
Here is the log: 
When using Neoforge 21.10.37-beta: crash-2025-11-05_11.28.10-client.txt: https://gnomebot.dev/paste/mclogs/q5XwKon
crash-2025-10-26_23.20.57-client.txt: https://gnomebot.dev/paste/mclogs/03RIhcH
latest.log: https://gnomebot.dev/paste/mclogs/BKpoeXv

`[461](https://gnomebot.dev/paste/mclogs/BKpoeXv#L461)    [26Oct2025 23:20:54.060] [modloading-worker-0/ERROR] [net.neoforged.fml.javafmlmod.FMLModContainer/LOADING]: Failed to register automatic subscribers. ModID: architectury
[462](https://gnomebot.dev/paste/mclogs/BKpoeXv#L462)    java.lang.NoClassDefFoundError: net/neoforged/neoforge/client/event/ContainerScreenEvent$Render$Background
[463](https://gnomebot.dev/paste/mclogs/BKpoeXv#L463)    	at java.lang.Class.getDeclaredMethods0:native
[464](https://gnomebot.dev/paste/mclogs/BKpoeXv#L464)    	at java.lang.Class.privateGetDeclaredMethods:L3578
[465](https://gnomebot.dev/paste/mclogs/BKpoeXv#L465)    	at java.lang.Class.getDeclaredMethods:L2676
[466](https://gnomebot.dev/paste/mclogs/BKpoeXv#L466)    	at net.neoforged.fml.javafmlmod.AutomaticEventSubscriber.lambda$inject$4:L56
[467](https://gnomebot.dev/paste/mclogs/BKpoeXv#L467)    	at java.util.ArrayList.forEach:L1596
[468](https://gnomebot.dev/paste/mclogs/BKpoeXv#L468)    	at net.neoforged.fml.javafmlmod.AutomaticEventSubscriber.inject:L47
[469](https://gnomebot.dev/paste/mclogs/BKpoeXv#L469)    	at net.neoforged.fml.javafmlmod.FMLModContainer.constructMod:L129
[470](https://gnomebot.dev/paste/mclogs/BKpoeXv#L470)    	at net.neoforged.fml.ModLoader.lambda$constructMods$1:L130
[471](https://gnomebot.dev/paste/mclogs/BKpoeXv#L471)    	at net.neoforged.fml.ModLoader.lambda$dispatchParallelTask$5:L208
[472](https://gnomebot.dev/paste/mclogs/BKpoeXv#L472)    	at java.util.concurrent.CompletableFuture.uniHandle:L934
[473](https://gnomebot.dev/paste/mclogs/BKpoeXv#L473)    	at java.util.concurrent.CompletableFuture$UniHandle.tryFire:L911
[474](https://gnomebot.dev/paste/mclogs/BKpoeXv#L474)    	at java.util.concurrent.CompletableFuture$Completion.exec:L483
[475](https://gnomebot.dev/paste/mclogs/BKpoeXv#L475)    	at java.util.concurrent.ForkJoinTask.doExec:L387
[476](https://gnomebot.dev/paste/mclogs/BKpoeXv#L476)    	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec:L1312
[477](https://gnomebot.dev/paste/mclogs/BKpoeXv#L477)    	at java.util.concurrent.ForkJoinPool.scan:L1843
[478](https://gnomebot.dev/paste/mclogs/BKpoeXv#L478)    	at java.util.concurrent.ForkJoinPool.runWorker:L1808
[479](https://gnomebot.dev/paste/mclogs/BKpoeXv#L479)    	at java.util.concurrent.ForkJoinWorkerThread.run:L188
[480](https://gnomebot.dev/paste/mclogs/BKpoeXv#L480)    Caused by: java.lang.ClassNotFoundException: net.neoforged.neoforge.client.event.ContainerScreenEvent$Render$Background
[481](https://gnomebot.dev/paste/mclogs/BKpoeXv#L481)    	at net.neoforged.fml.classloading.ModuleClassLoader.loadClass:L254
[482](https://gnomebot.dev/paste/mclogs/BKpoeXv#L482)    	at java.lang.ClassLoader.loadClass:L526`

--- a/gradle.properties
+++ b/gradle.properties
@@ -19,7 +19,7 @@ fabric_api_version=0.136.0+1.21.10
 mod_menu_version=15.0.0-beta.1
 
 forge_version=51.0.0
-neoforge_version=21.10.32-beta
+neoforge_version=21.10.38-beta
 
 # Set to empty if not snapshots
 neoforge_pr=

--- a/neoforge/src/main/java/dev/architectury/event/forge/EventHandlerImplClient.java
+++ b/neoforge/src/main/java/dev/architectury/event/forge/EventHandlerImplClient.java
@@ -143,11 +143,6 @@ public class EventHandlerImplClient {
     }
     
     @SubscribeEvent(priority = EventPriority.HIGH)
-    public static void eventContainerScreenEvent(ContainerScreenEvent.Render.Background event) {
-        ClientGuiEvent.RENDER_CONTAINER_BACKGROUND.invoker().render(event.getContainerScreen(), event.getGuiGraphics(), event.getMouseX(), event.getMouseY(), Minecraft.getInstance().getDeltaTracker().getRealtimeDeltaTicks());
-    }
-    
-    @SubscribeEvent(priority = EventPriority.HIGH)
     public static void eventContainerScreenEvent(ContainerScreenEvent.Render.Foreground event) {
         ClientGuiEvent.RENDER_CONTAINER_FOREGROUND.invoker().render(event.getContainerScreen(), event.getGuiGraphics(), event.getMouseX(), event.getMouseY(), Minecraft.getInstance().getDeltaTracker().getRealtimeDeltaTicks());
     }

--- a/neoforge/src/main/java/dev/architectury/mixin/forge/client/MixinAbstractContainerScreen.java
+++ b/neoforge/src/main/java/dev/architectury/mixin/forge/client/MixinAbstractContainerScreen.java
@@ -1,0 +1,25 @@
+package dev.architectury.mixin.forge.client;
+
+import dev.architectury.event.events.client.ClientGuiEvent;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.network.chat.Component;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(AbstractContainerScreen.class)
+public abstract class MixinAbstractContainerScreen extends Screen {
+    protected MixinAbstractContainerScreen(Component component) {
+        super(component);
+    }
+    
+    @Inject(method = "renderBackground",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screens/inventory/AbstractContainerScreen;renderBg(Lnet/minecraft/client/gui/GuiGraphics;FII)V",
+                    ordinal = 0, shift = At.Shift.AFTER))
+    public void renderBackground(GuiGraphics graphics, int mouseX, int mouseY, float delta, CallbackInfo ci) {
+        ClientGuiEvent.RENDER_CONTAINER_BACKGROUND.invoker().render((AbstractContainerScreen<?>) (Object) this, graphics, mouseX, mouseY, delta);
+    }
+}

--- a/neoforge/src/main/resources/architectury.mixins.json
+++ b/neoforge/src/main/resources/architectury.mixins.json
@@ -7,6 +7,7 @@
   "client": [
     "client.MixinClientPacketListener",
     "client.MixinCommandSourceStack",
+    "client.MixinAbstractContainerScreen",
     "MixinClientLevel",
     "MixinMinecraft"
   ],

--- a/testmod-common/src/main/java/dev/architectury/test/TestModClient.java
+++ b/testmod-common/src/main/java/dev/architectury/test/TestModClient.java
@@ -21,6 +21,7 @@ package dev.architectury.test;
 
 import com.mojang.brigadier.arguments.StringArgumentType;
 import dev.architectury.event.events.client.ClientCommandRegistrationEvent;
+import dev.architectury.event.events.client.ClientGuiEvent;
 import dev.architectury.event.events.client.ClientLifecycleEvent;
 import dev.architectury.registry.client.gui.ClientTooltipComponentRegistry;
 import dev.architectury.registry.client.level.entity.EntityRendererRegistry;
@@ -29,9 +30,13 @@ import dev.architectury.test.registry.TestRegistries;
 import dev.architectury.test.registry.client.TestKeybinds;
 import dev.architectury.test.registry.objects.ItemWithTooltip;
 import dev.architectury.test.registry.objects.MyClientTooltipComponent;
+import net.minecraft.client.gui.components.FocusableTextWidget;
 import net.minecraft.client.renderer.entity.CowRenderer;
 
 import static dev.architectury.test.TestMod.SINK;
+
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
 
 public class TestModClient {
     public static void initializeClient() {
@@ -51,5 +56,8 @@ public class TestModClient {
                                 return 0;
                             })));
         });
+        ClientGuiEvent.RENDER_CONTAINER_BACKGROUND.register(((screen, graphics, mouseX, mouseY, delta) -> {
+            graphics.renderItem(new ItemStack(Items.DIAMOND), mouseX, mouseY);
+        }));
     }
 }

